receiver/sqlserverreceiver/scraper.go

```diff
@@ -573,110 +575,143 @@ func (s *sqlServerScraperHelper) recordDatabaseQueryTextAndPlan(ctx context.Cont
+				valueSetter:    setInt,
+			},
+			{
+				key:            dbPrefix + totalElapsedTime,
+				valueRetriever: defaultValueRetriever(float64(totalElapsedTimeDiffsMicroSecond[i]) / 1000 / 1000),
+				valueSetter:    setDouble,
+			},
+			{
+				key:            dbPrefix + totalGrant,
+				columnName:     totalGrant,
+				valueRetriever: retrieveInt,
+				valueSetter:    setInt,
+			},
+			{
+				key:            "db.system.name",
+				valueRetriever: defaultValueRetriever("microsoft.sql_server"),
+				valueSetter:    setString,
+			},
+			{
+				key:            instanceNameKey,
+				columnName:     instanceNameKey,
+				valueRetriever: vanillaRetriever,
+				valueSetter:    setString,
+			},
+			{
+				key:            serverAddressKey,
+				valueRetriever: defaultValueRetriever(s.config.Server),
+				valueSetter:    setString,
+			},
+			{
+				key:            serverPortKey,
+				valueRetriever: defaultValueRetriever((int64(s.config.Port))),
+				valueSetter:    setInt,
+			},
 		}
 
-		// handling `total_logical_writes`
-		logicalWritesVal, err := strconv.ParseInt(row[logicalWrites], 10, 64)
-		if err != nil {
-			err = fmt.Errorf("row %d: %w", i, err)
-			errs = append(errs, err)
-		}
-		if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, logicalWrites, logicalWritesVal); cached {
-			record.Attributes().PutInt(dbPrefix+logicalWrites, diff)
+		updatedOnly := map[string]bool{
+			rowsReturned:   true,
+			logicalReads:   true,
+			logicalWrites:  true,
+			physicalReads:  true,
+			executionCount: true,
+			totalGrant:     true,
 		}
 
-		// handling `physical_reads`
-		physicalReadsVal, err := strconv.ParseInt(row[physicalReads], 10, 64)
-		if err != nil {
-			err = fmt.Errorf("row %d: %w", i, err)
-			errs = append(errs, err)
-		}
-		if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, physicalReads, physicalReadsVal); cached {
-			record.Attributes().PutInt(dbPrefix+physicalReads, diff)
-		}
+		s.logger.Debug(fmt.Sprintf("QueryHash: %v, PlanHash: %v, DataRow: %v", queryHashVal, queryPlanHashVal, row))
 
-		// handling `execution_count`
-		totalExecutionCount, err := strconv.ParseInt(row[executionCount], 10, 64)
+		// handle `total_worker_time`, storing seconds
+		// it is a little bit tricky to put this to the array based workflow,
+		// as the value need to be divided -> type assertion -> check cache.
+		// hence handle it separately.
+		workerTimeMicroSecond, err := strconv.ParseInt(row[totalWorkerTime], 10, 64)
 		if err != nil {
 			err = fmt.Errorf("row %d: %w", i, err)
 			errs = append(errs, err)
 		} else {
-			if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, executionCount, totalExecutionCount); cached {
-				record.Attributes().PutInt(dbPrefix+executionCount, diff)
+			if cached, diffMilliSecond := s.cacheAndDiff(queryHashVal, queryPlanHashVal, totalWorkerTime, workerTimeMicroSecond/1000); cached {
+				record.Attributes().PutDouble(dbPrefix+totalWorkerTime, float64(diffMilliSecond)/1000)
```

Comment:
```suggestion
			if cached, diffMicrosecond := s.cacheAndDiff(queryHashVal, queryPlanHashVal, totalWorkerTime, workerTimeMicroSecond); cached {
				record.Attributes().PutDouble(dbPrefix+totalWorkerTime, float64(diffMicrosecond)/1000_000)
```

Could we keep the original unit until `PutDouble` and not introduce a new unit, `millisecond`, here?